src/libraries/System.Private.CoreLib/src/System/IO/Strategies/WindowsFileStreamStrategy.cs

```diff
@@ -120,6 +130,8 @@ internal sealed override SafeFileHandle SafeFileHandle
                     // in memory position is out-of-sync with the actual file position.
                     FileStreamHelpers.Seek(_fileHandle, _path, _filePosition, SeekOrigin.Begin);
                 }
+                _exposed = true;
+                _length = -1; // invalidate cached length
```

Comment:
I think this is not required as long as we always access `Length` rather than this field directly.

Reply:
You are right, it's not required. I just wanted to avoid situation where someone is debugging and can observe `_length` and `Length` having some positive, but different values.